### PR TITLE
MC-9236: RBAC 2

### PR DESCRIPTION
**Pull request by franzpgarcia**

### CODE REVIEW

File: source/includes/administration/_additional_roles.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+## Additional roles
+
+With additional roles, you can allow users access to other organizations. You can specify a scope which targets one or many organizations with a specific role. See [roles](#administration-roles) for more information on roles.
```
Comment (simongodard) [inline source/includes/administration/_additional_roles.md:3]: The word `to` is not needed access and other

File: source/includes/administration/_additional_roles.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+### Scoping
+
+There are different types of scope that you can use depending on your use case. This scope must be provided when assigning an additional role to a user. You cannot give a scope/role combination to another user if you don't have a higher or equivalent additional role.
```
Comment (simongodard) [inline source/includes/administration/_additional_roles.md]: There are different types of scope`s`...

File: source/includes/administration/_additional_roles.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+ORG_TOPLEVEL | Applies the role on all top-level organizations | &nbsp;
+TAGS_ANYMATCH | Applies the role on all organizations that are tagged with at least one the specified tags | `tags`
+GLOBAL | Applies role on all organizations | &nbsp;
+
+### Get additional roles of user
+
+`GET /users/:id/additional_roles`
+
+```shell
+# Update an environment
```
Comment (simongodard) [inline source/includes/administration/_additional_roles.md]: Wrong comment

File: source/includes/administration/_additional_roles.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    }]
+}
+```
+
+Attributes | &nbsp;
+---------- | -----------
+`id`<br/>*UUID* | The id of the additional role
+`creationDate`<br/>*string* | The date the additional role was added/updated on the user
+`scopeQualifier`<br/>*string* | The scope of this additional role (ORG_BASE, TAGS_ANYMATCH, GLOBAL, etc..). Please refer to the table above.
+`tags`<br/>*[string]* | Organization tags to match for TAGS_ANYMATCH scoping
```
Comment (simongodard) [inline source/includes/administration/_additional_roles.md:64]: Should we mention tags should be comma-separated?

File: source/includes/administration/_additional_roles.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+`organization`<br/>*[Organization](#administration-organizations)* | The organization of the additional role. Only present on scopes that target a specific organization.<br/>*includes*: `id`, `name`, `entryPoint`
+`user`<br/>*[User](#administration-users)* | The user that the additional role is applied on<br/>*includes*: `id`, `username`, `firstName`, `lastName`, `email`
+
+
+### Assign additional role to user
+
+`POST /users/:id/additional_roles`
+
+```shell
+# Update an environment
```
Comment (simongodard) [inline source/includes/administration/_additional_roles.md]: Wrong comment here too.

File: source/includes/administration/_additional_roles.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+```
+
+Required | &nbsp;
+---------- | -----------
+`scopeQualifier`<br/>*string* | The scope of this additional role (ORG_BASE, TAGS_ANYMATCH, GLOBAL, etc..)
+`role.id`<br/>*UUID* | The id of the role to apply on the target scope
+`organization.id`<br/>*UUID* | The id of the target organization of the scope. Only required when using a scope qualifier that targets an organization.
+`tags`<br/>*UUID* | The organization tags of the scope. Only required when using the TAGS_ANYMATCH scope.
+
+You will need to have `Admin users` permission to execute this operation.
```
Comment (simongodard) [inline source/includes/administration/_additional_roles.md]: How should we present the permissions names?:
1. `Admin users`
2. `Admin:users`
3. `Manage users`

Comment (franzpgarcia) [reply source/includes/administration/_additional_roles.md]: good question ;S

File: source/includes/administration/_additional_roles.md (hunk 4 above)
Comment (franzpgarcia) [reply source/includes/administration/_additional_roles.md:64]: It's actually a list

File: source/includes/administration/_additional_roles.md (hunk 5 above)
Comment (franzpgarcia) [reply source/includes/administration/_additional_roles.md]: fixed

File: source/includes/administration/_additional_roles.md (hunk 6 above)
Comment (simongodard) [reply source/includes/administration/_additional_roles.md]: I think it should be the user label.
image

so:
4. `Users : Manage`

Comment (simongodard) [reply source/includes/administration/_additional_roles.md]: @mavailla just confirmed it should be the user label.

File: source/includes/administration/_additional_roles.md (hunk 2 above)
Comment (pdube) [inline source/includes/administration/_additional_roles.md]: The double negative in the last sentence is a bit confusing. How about something like : `You can delegate a lesser or equivalent role binding that you have`

File: source/includes/administration/_additional_roles.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+"data": [
+    {
+        "id": "ed18cdd0-0fa0-464d-a993-e689d39dfe2e",
+        "creationDate": "2019-06-06T15:01:22.000Z",
+        "scopeQualifier": "ORG_BASE",
+        "role": {
+            "id": "cc71a40b-d90b-40db-b724-08d71f5179b5",
+            "name": "user",
+            "isSystem": true,
+            "isFixed": true
```
Comment (pdube) [inline source/includes/administration/_additional_roles.md:41]: Should we rename these to just be `system` and `fixed`?

**Review by jasondias9 — APPROVED** [review]

File: source/includes/administration/_additional_roles.md (hunk 6 above)
Comment (pdube) [reply source/includes/administration/_additional_roles.md]: +1

**Review by pdube — COMMENTED** [review]
Should we update the `users` endpoint to be v2 right now? Nothing has changed yet

File: source/includes/administration/_additional_roles.md (hunk 7 above)
Comment (franzpgarcia) [reply source/includes/administration/_additional_roles.md:41]: Not really doc related but would make sense.

Comment (pdube) [reply source/includes/administration/_additional_roles.md:41]: Yeah, not related to this PR, but it struck me when I was reading the doc

File: source/includes/administration/_additional_roles.md (hunk 6 above)
Comment (franzpgarcia) [reply source/includes/administration/_additional_roles.md]: Fixed

File: source/includes/administration/_additional_roles.md (hunk 2 above)
Comment (franzpgarcia) [reply source/includes/administration/_additional_roles.md]: Fixed

**Review by simongodard — APPROVED** [review]

**Pull request merged**